test/e2e/simple.go

```diff
@@ -50,10 +50,8 @@ func E2ESimple(logger *log.Logger) error {
 	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
 	defer cancel()
 	opts := txsim.DefaultOptions().WithSeed(seed).SuppressLogs()
-	err = txsim.Run(ctx, testnet.GRPCEndpoints()[0], kr, encCfg, opts, sequences...)
-
-	if !errors.Is(err, context.DeadlineExceeded) {
```

Comment:
Does this not work because the error gets wrapped? If that is the case I would prefer that the error gets unwrapped / or otherwise handled in txsim itself

Reply:
I think a context deadline exceeded error gets wrapped in an RPC error but TBH then I would expect the original implementation `errors.Is` to have worked.

I can hold off on merging this b/c I don't have a convincing way to prove this helped or hurt b/c the test is flaky for other reasons outside this issue.

Reply:
Yeah I think we should then unwrap it in txsim and just return the raw deadline exceeded error